docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+  ``-_.``
+
+* must not be a string composed only of period characters, e.g. "." or
+  ".."
+
+* must be at most 255 characters long
+
+Node names are case sensitive, e.g., the names "foo" and "FOO" are **not**
+identical.
+
+.. note:
+    The Zarr core development team recognises that restricting the set
+    of allowed characters creates an impediment and bias against users
+    of different languages. We are actively discussing whether the full
+    Unicode character set could be allowed and what technical issues
+    this would entail. If you have experience or views please comment on
+    `issue #56 <https://github.com/zarr-developers/zarr-specs/issues/56>`_.
+
+
+Data types
+==========
+
+A data type describes the set of possible binary values that an array
+element may take, along with some information about how the values
+should be interpreted.
+
+This core specification defines a limited set of data types to
+represent boolean values, integers, and floating point
+numbers. Extensions may define additional data types. All of the data
+types defined here have a fixed size, in the sense that all values
+require the same number of bytes. However, extensions may define
+variable sized data types.
+
+Note that the Zarr specification is intended to enable communication
+of data between a variety of computing environments. The native byte
+order may differ between machines used to write and read the data.
+
+Each data type is associated with an identifier, which can be used in
+metadata documents to refer to the data type. For the data types
+defined in this specification, the identifier is a simple ASCII
+string. However, extensions may use any JSON value to identify a data
+type.
+
+
+Core data types
+---------------
+
+.. list-table:: Data types
+   :header-rows: 1
+
+   * - Identifier
+     - Numerical type
+     - Size (no. bytes)
+     - Byte order
+   * - ``bool``
+     - Boolean, with False encoded as ``\\x00`` and True encoded as ``\\x01``
+     - 1
+     - None
+   * - ``i1``
+     - signed integer
+     - 1
+     - None
+   * - ``<i2``
+     - signed integer
+     - 2
+     - little-endian
+   * - ``<i4``
+     - signed integer
+     - 4
+     - little-endian
+   * - ``<i8``
+     - signed integer
+     - 8
+     - little-endian
+   * - ``>i2``
+     - signed integer
+     - 2
+     - big-endian
+   * - ``>i4``
+     - signed integer
+     - 4
+     - big-endian
+   * - ``>i8``
+     - signed integer
+     - 8
+     - big-endian
+   * - ``u1``
+     - unsigned integer
+     - 1
+     - None
+   * - ``<u2``
+     - unsigned integer
+     - 2
+     - little-endian
+   * - ``<u4``
+     - unsigned integer
+     - 4
+     - little-endian
+   * - ``<u8``
+     - unsigned integer
+     - 8
+     - little-endian
+   * - ``>u2``
+     - unsigned integer
+     - 2
+     - big-endian
+   * - ``>u4``
+     - unsigned integer
+     - 4
+     - big-endian
+   * - ``>u8``
+     - unsigned integer
+     - 8
+     - big-endian
+   * - ``<f2``
```

Comment:
Many programming languages don't natively support float16, so it might make sense to note that this is optionally supported.

Reply:
resolved via #155